lib/ansible/modules/cloud/vmware/vmware_vcenter_permission.py

```diff
@@ -0,0 +1,359 @@
+            - describes user to set permission to
+        required: true
+        type: dict
+        options:
+            name:
+                description:
+                    - name of user
+                required: true
+                type: str
+            domain:
+                description:
+                    - domain of user
+                required: true
+                type: str
+            isGroup:
+                description:
+                    - indicates if user is a group
+                default: false
+                type: bool
+    role_name:
+        description:
+            - Name of the role to assign
+            - required if state is present
+        required: false
+        type: str
+    propagate:
+        description:
+            - set to true if permisison should be propagated to children
+        required: false
+        type: bool
+        default: false
+    state:
+        description:
+            - set to present to add/update permisisons, absent to delete them
+        required: false
+        type: str
+        choices: ['present', 'absent']
+        default: 'present'
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: add or update permission on datacenter DC
+  vmware_vcenter_permission:
+    hostname: "{{ vCenter_hostname }}"
+    username: "{{ vCenter_username }}"
+    password: "{{ vCenter_password }}"
+    validate_certs: False
+    role_name: 'VM-Power'
+    entity:
+          name: 'DC'
+          type: 'Datacenter'
+    principal:
+          name: 'test'
+          domain: 'vsphere.local'
+          isGroup: False
+    propagate: True
+    state: present
+
+- name: remove permission from datacenter DC
+  vmware_vcenter_permission:
+    hostname: "{{ vCenter_hostname }}"
+    username: "{{ vCenter_username }}"
+    password: "{{ vCenter_password }}"
+    validate_certs: False
+    role_name: 'VM-Power'
+    entity:
+          name: 'DC'
+          type: 'Datacenter'
+    principal:
+          name: 'test'
+          domain: 'vsphere.local'
+          isGroup: False
+    propagate: True
+    state: absent
+'''
+
+RETURN = '''
+
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (HAS_PYVMOMI,
+                                         connect_to_api,
+                                         vmware_argument_spec,
+                                         find_object_by_name
+                                         )
+
+class VmwareVcenterPermission(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.state = self.module.params['state']
+
+        self.content = connect_to_api(module)
+        self.authMgr = self.content.authorizationManager
+
+        if self.state == 'present':
+            self.role_name = self.module.params['role_name']
```

Comment:
You don't need use `self.role_name` after in the module, you don't need to define it as an attribute.